sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -602,6 +603,16 @@ object CombineLimits extends Rule[LogicalPlan] {
 }
 
 /**
+ * Combines two adjacent [[Cast]] expressions into one
+ */
+object CombineCasts extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan transformAllExpressions {
+    case cc @ Cast(c2 @ Cast(c1, dataType1), dataType2) =>
```

Comment:
`cc` and `c2` is not used. You could just write `case Cast(Cast(c1, dataType1), dataType2)`